common/transport/amqp/src/amqp.ts

```diff
@@ -34,12 +31,90 @@ export type GenericAmqpBaseCallback<T> = (err: Error | null, result?: T) => void
  * @param   {String}    sdkVersionString        String identifying the type and version of the SDK used.
  */
 export class Amqp {
-  private _amqp: amqp10.Client;
+  //
+  // Top of the rhea object hierarchy.
+  //
+  private _rheaContainer: Container;
+  //
+  // Rhea object used to represent the connection.  The connection object is a child of the container object.
+  // It is an event emitter.  Numerous handlers listen to manage setting up a connection and act upon error events
+  // associated with the connection.
+  //
+  private _rheaConnection: Connection;
```

Comment:
>_rheaConnection [](start = 10, length = 15)

- aren't the connection and the session part of the container or available in a context object?
- is there a risk that we are keeping references to things that the container has already gotten rid of?

Reply:
Don't believe that there is a direct property from the container to the connection.

As for the context,  You only have those when handling and error.

---
In reply to: [209127965](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209127965) [](ancestors = 209127965)